### PR TITLE
ONe attachment test refactor

### PR DESCRIPTION
**Pull request by pauloewerton**
This change refactors ONe attachment plugin as well as its unit tests. Also, a base tests class for ONe tests is added. To test this PR: check out code; configure an ONe cloud in your RAS environment; test attachment operations via a client; run attachment unit tests.

### CODE REVIEW

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/opennebula/attachment/v5_4/OpenNebulaAttachmentPluginTest.java [hunk 1]
```diff
@@ -19,72 +19,61 @@
...
-
-	private static final String CLOUD_NAME = "opennebula";
-	private static final String DEFAULT_DEVICE_PREFIX = "vd";
-	private static final String FAKE_DEVICE = "fake-image-device";
-	private static final String FAKE_ID_PROVIDER = "fake-id-provider";
-	private static final String FAKE_INSTANCE_ID = "1";
-	private static final String FAKE_NAME = "fake-name";
-	private static final String FAKE_PROVIDER = "fake-provider";
-	private static final String FAKE_USER_ID = "fake-user-id";
+@PrepareForTest({DatabaseManager.class, OpenNebulaClientUtil.class, SharedOrderHolders.class, VirtualMachine.class})
```
Comment (wassoares) [inline src/test/java/cloud/fogbow/ras/core/plugins/interoperability/opennebula/attachment/v5_4/OpenNebulaAttachmentPluginTest.java]: **SharedOrderHolders**.class can be removed from this line as there is no more mock static for this class.

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/opennebula/attachment/v5_4/OpenNebulaAttachmentPluginTest.java [hunk 2]
```diff
@@ -150,277 +139,349 @@ public void testHasFailedUnsuccessful() {
...
-		Mockito.when(virtualMachine.info()).thenReturn(response);
-		Mockito.when(response.getMessage()).thenReturn(VIRTUAL_MACHINE_CONTENT);
-
-		CloudUser cloudUser = createCloudUser();
+		Mockito.when(this.virtualMachine.diskAttach(this.template)).thenReturn(this.response);
+		Mockito.when(this.response.isError()).thenReturn(false);
 
 		// exercise
-		this.plugin.requestInstance(attachmentOrder, cloudUser);
+		this.plugin.requestInstance(this.attachmentOrder, this.cloudUser);
```
Comment (ANKerD) [inline src/test/java/cloud/fogbow/ras/core/plugins/interoperability/opennebula/attachment/v5_4/OpenNebulaAttachmentPluginTest.java]: this test should mock doRequestInstance() and getDiskIdFromContentOf() and below, create separate tests for getDiskIdFromContentOf().

Comment (ANKerD) [reply src/test/java/cloud/fogbow/ras/core/plugins/interoperability/opennebula/attachment/v5_4/OpenNebulaAttachmentPluginTest.java]: Same for deleteInstance and getInstance related methods.

File: src/main/java/cloud/fogbow/ras/core/plugins/interoperability/opennebula/attachment/v5_4/OpenNebulaAttachmentPlugin.java [hunk 3]
```diff
@@ -55,23 +53,23 @@ public boolean hasFailed(String cloudState) {
...
+	public String requestInstance(AttachmentOrder order, CloudUser cloudUser) throws FogbowException {
+		// NOTE(pauloewerton): expecting a default target device such as /dev/sdb
+		String[] targetList = order.getDevice().split(DEVICE_PATH_SEPARATOR);
 		String target = targetList.length == 3 ? targetList[TARGET_INDEX] : DEFAULT_TARGET;
+		String imageId = order.getVolumeId();
 
 		CreateAttachmentRequest request = new CreateAttachmentRequest.Builder()
 				.imageId(imageId)
                 .target(target)
 				.build();
```
[the remark this answers is not in the document]
Comment (pauloewerton) [reply src/main/java/cloud/fogbow/ras/core/plugins/interoperability/opennebula/attachment/v5_4/OpenNebulaAttachmentPlugin.java]: can you please check if the indentation appears wrong on your ide? it looks fine on my local copy, so maybe this is some sort of github glitch ;-)

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/opennebula/attachment/v5_4/OpenNebulaAttachmentPluginTest.java (hunk 1 above)
Comment (pauloewerton) [reply src/test/java/cloud/fogbow/ras/core/plugins/interoperability/opennebula/attachment/v5_4/OpenNebulaAttachmentPluginTest.java]: done.

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/opennebula/attachment/v5_4/OpenNebulaAttachmentPluginTest.java [hunk 4]
```diff
@@ -19,72 +19,61 @@
...
 import cloud.fogbow.ras.core.models.orders.ComputeOrder;
-import cloud.fogbow.ras.core.models.orders.OrderState;
 import cloud.fogbow.ras.core.models.orders.VolumeOrder;
 import cloud.fogbow.ras.core.plugins.interoperability.opennebula.OpenNebulaClientUtil;
 import cloud.fogbow.ras.core.plugins.interoperability.opennebula.OpenNebulaStateMapper;
 
+import static cloud.fogbow.ras.core.plugins.interoperability.opennebula.attachment.v5_4.OpenNebulaAttachmentPlugin.IMAGE_ID_PATH_FORMAT;
+import static cloud.fogbow.ras.core.plugins.interoperability.opennebula.attachment.v5_4.OpenNebulaAttachmentPlugin.TARGET_PATH_FORMAT;
+
 @RunWith(PowerMockRunner.class)
```
[the remark this answers is not in the document]
Comment (pauloewerton) [reply src/test/java/cloud/fogbow/ras/core/plugins/interoperability/opennebula/attachment/v5_4/OpenNebulaAttachmentPluginTest.java]: good catch!

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/opennebula/attachment/v5_4/OpenNebulaAttachmentPluginTest.java (hunk 2 above)
Comment (pauloewerton) [reply src/test/java/cloud/fogbow/ras/core/plugins/interoperability/opennebula/attachment/v5_4/OpenNebulaAttachmentPluginTest.java]: done.

**Comment by pauloewerton** [comment]
thanks @ANKerD and @wassoares for your review. comments addressed.

**Pull request merged**